hawkular-alerts-api/src/main/java/org/hawkular/alerts/api/model/event/Alert.java

```diff
@@ -63,6 +63,9 @@
     private List<Note> notes;
 
     @JsonInclude(Include.NON_EMPTY)
+    private List<LifeCycle> lifecycle;
```

Comment:
> In the model we might have open -> ack -> resolved -> ack -> resolved

I'm not aware of this being valid or at all common.  This sort of lifecycle support has not been a goal for us, and actually, I believe there is code to actively prevent going back to Ack from Resolved.

> At the moment that we index resolved/ack dates for filtering with an auxiliar table we need to track the state to maintain this.

I'm not sure I understand this comment.  Yes, we would have to index the fields but we already have that pattern for a variety of fields, no?   If we are to add Lifecycle then I wonder the same as @moolitayer, why keep the explicit ack/resolve fields?